### PR TITLE
Add download logs button component and implement it in dashboard

### PR DESCRIPTION
**Pull request by Nikeleos**

### CODE REVIEW

File: src/Dashboard/components/ButtonDownloadLogs/ButtonDownloadLogs.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+          </Button>
+        </Grid>
+      </Grid>
+    </div>
+  );
+};
+
+ButtonDownloadLogs.propTypes = {
+  download: PropTypes.func.isRequired,
+  laebl: PropTypes.string.isRequired
```
Comment (csm-thu) [inline src/Dashboard/components/ButtonDownloadLogs/ButtonDownloadLogs.js:37]: Typo here

File: src/Dashboard/components/ButtonDownloadLogs/ButtonDownloadLogs.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Grid, Button, makeStyles } from '@material-ui/core';
+
+const useStyles = makeStyles(theme => ({
+  root: {
```
Comment (csm-thu) [inline src/Dashboard/components/ButtonDownloadLogs/ButtonDownloadLogs.js:9]: The "root" class is not used

File: src/Dashboard/components/ButtonDownloadLogs/ButtonDownloadLogs.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  const classes = useStyles();
+  const {
+    download,
+    label
+  } = props;
+  
+  return (
+    <div>
+      <Grid container spacing={3} direction="row" justify="flex-start" alignItems="center"
+        className={classes.gridContainer}>
```
Comment (csm-thu) [inline src/Dashboard/components/ButtonDownloadLogs/ButtonDownloadLogs.js:24]: The "gridContainer" class is not defined in useStyles

File: src/Dashboard/components/ButtonDownloadLogs/ButtonDownloadLogs.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+const ButtonDownloadLogs = (props) => {
+  const classes = useStyles();
+  const {
+    download,
+    label
+  } = props;
+  
+  return (
+    <div>
+      <Grid container spacing={3} direction="row" justify="flex-start" alignItems="center"
```
Comment (csm-thu) [inline src/Dashboard/components/ButtonDownloadLogs/ButtonDownloadLogs.js:23]: Do we need to use a Grid for a single item ?

File: src/Dashboard/Dashboard.js [hunk 5]
```diff
@@ -24,6 +24,7 @@ const Dashboard = (props) => {
     scenarioId,
     csmSimulationRun,
     scenarioState,
+    downloadScenarioRunCumulatedLogsFile,
```
Comment (csm-thu) [inline src/Dashboard/Dashboard.js:27]: Could we use a shorter name for this function in Dashboard and DashboardPlaceholder ? "downloadLogs" seems understandable enough to me :)

**Review by csm-thu — CHANGES_REQUESTED** [review]

**Review by jreynard-code — CHANGES_REQUESTED** [review]
Nothing to add (thanks @csm-thu  <3 )

File: src/Dashboard/components/ButtonDownloadLogs/ButtonDownloadLogs.js (hunk 1 above)
Comment (Nikeleos) [reply src/Dashboard/components/ButtonDownloadLogs/ButtonDownloadLogs.js:37]: Done

File: src/Dashboard/components/ButtonDownloadLogs/ButtonDownloadLogs.js (hunk 2 above)
Comment (Nikeleos) [reply src/Dashboard/components/ButtonDownloadLogs/ButtonDownloadLogs.js:9]: Done

File: src/Dashboard/components/ButtonDownloadLogs/ButtonDownloadLogs.js (hunk 4 above)
Comment (Nikeleos) [reply src/Dashboard/components/ButtonDownloadLogs/ButtonDownloadLogs.js:23]: With a view to upgrade the component in the future, it is more convenient to implement directly a Grid who already possess all the display parameters.

File: src/Dashboard/Dashboard.js (hunk 5 above)
Comment (Nikeleos) [reply src/Dashboard/Dashboard.js:27]: Done

File: src/Dashboard/components/ButtonDownloadLogs/ButtonDownloadLogs.js (hunk 3 above)
Comment (Nikeleos) [reply src/Dashboard/components/ButtonDownloadLogs/ButtonDownloadLogs.js:24]: Done

File: src/Dashboard/Dashboard.js [hunk 6]
```diff
@@ -34,8 +35,8 @@ const Dashboard = (props) => {
   // Handle optional status property
   const noRun = scenarioState === 'Created' || scenarioState === null;
   const runInProgress = scenarioState === 'Running';
-  const hasError = scenarioState === 'Failed';
-  const isReady = scenarioState === undefined || scenarioState === 'Successful';
+  const hasError = scenarioState === 'Successful';
+  const isReady = scenarioState === undefined || scenarioState === 'Failed';
```
Comment (csm-thu) [inline src/Dashboard/Dashboard.js:39]: Something is odd with these variables: the version before the change (`hasError = scenarioState === 'Failed';`) seems to be better than after the change (`hasError = scenarioState === 'Successful';`)

File: src/Dashboard/components/DashboardPlaceholder/DashboardPlaceholder.js [hunk 7]
```diff
@@ -25,10 +26,10 @@ const useStyles = makeStyles(theme => ({
 const DashboardPlaceholder = (props) => {
   const classes = useStyles();
   const { t } = useTranslation();
-  const { labelKey, defaultLabel, icon } = props;
+  const { labelKey, defaultLabel, icon, button, downloadCumulatedLogs } = props;
```
Comment (csm-thu) [inline src/Dashboard/components/DashboardPlaceholder/DashboardPlaceholder.js:29]: The "button" property should be removed here too (not used, and not declared in the propTypes)

File: src/Dashboard/Dashboard.js (hunk 6 above)
Comment (Nikeleos) [reply src/Dashboard/Dashboard.js:39]: It is a mistake. It is only to force and testing error diplay

File: src/Dashboard/components/DashboardPlaceholder/DashboardPlaceholder.js (hunk 7 above)
Comment (Nikeleos) [reply src/Dashboard/components/DashboardPlaceholder/DashboardPlaceholder.js:29]: Done

**Pull request merged**